### PR TITLE
Add Node Resolution advanced topic

### PR DESCRIPTION
**Pull request by KB4MDD**
This adds a Node Resolution topic to the advanced section.

### CODE REVIEW

File: docs/adv-topics/noderesolution.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Node Resolution
+app_rpt has three ways to resolve a node number to an IP address.
```
Comment (Allan-N) [inline docs/adv-topics/noderesolution.md:2]: Add a blank line after this sentence to separate the text from the list.

File: docs/adv-topics/noderesolution.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+app_rpt has three ways to resolve a node number to an IP address.  
+* An external node directory file.
+* Domain Name Service (DNS).
+* Statically entered in rpt.conf
+
+The above mechanisms are used to translate a node number into its respective public IP address.  Connections are made directly between nodes using their public IP addresses.
+
+app_rpt first looks to see if the node is statically configured.  If it is not found, the setting for **node_lookup_method** in the **[general]** stanza of rpt.conf determines the next step. The default setting is to perform a DNS lookup and if not found, attempt a lookup using the external file.  For more information see [HTTP Registration](https://allstarlink.github.io/adv-topics/httpreg/)
+
+*Note:  The AllStarLink network maintains security by requiring node owners to register with AllStarLink. Before a node is published to our directory, it must successfully authenticate and register with our server.*
```
Comment (Allan-N) [inline docs/adv-topics/noderesolution.md:11]: ... to our directory it must ...

File: docs/adv-topics/noderesolution.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+* Statically entered in rpt.conf
+
+The above mechanisms are used to translate a node number into its respective public IP address.  Connections are made directly between nodes using their public IP addresses.
+
+app_rpt first looks to see if the node is statically configured.  If it is not found, the setting for **node_lookup_method** in the **[general]** stanza of rpt.conf determines the next step. The default setting is to perform a DNS lookup and if not found, attempt a lookup using the external file.  For more information see [HTTP Registration](https://allstarlink.github.io/adv-topics/httpreg/)
+
+*Note:  The AllStarLink network maintains security by requiring node owners to register with AllStarLink. Before a node is published to our directory, it must successfully authenticate and register with our server.* 
+
+## External Node Directory File
+app_rpt can use an external node directory file to perform node resolution.  The ASL3-Update-NodeList package is used to periodically download the current list of authenticated nodes.
```
Comment (Allan-N) [inline docs/adv-topics/noderesolution.md:14]: ASL3-Update-NodeList package --> asl3-update-nodeList service

File: docs/adv-topics/noderesolution.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+[extnodes]
+
+2000=radio@162.248.93.134:4569/2000,162.248.93.134
+2001=radio@162.248.93.134:4569/2001,162.248.93.134
+```
+
+## Domain Name Service DNS ##
+AllStarLink now provides a DNS service for performing node resolution.  DNS is an Internet standard and is more efficient than using a file base solution.
+
+When using DNS app_rpt must know the length of the longest node number.  The current maximum is 6 digits.  This number can be overriden in rpt.conf by adding the **max_dns_node_length** key and value to the **[general]** stanza.
```
Comment (Allan-N) [inline docs/adv-topics/noderesolution.md:32]: overriden --> overridden

File: docs/adv-topics/noderesolution.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+For more information on DNS see [AllStarLink DNS Servers](https://wiki.allstarlink.org/wiki/DNS_Servers).
+
+## Statically Configured Node ##
+In some cases, it is desirable to statically define the node information.  This is commonly used for private nodes or private networks.
+
+To configure a static node, enter the information for te node in rpt.conf under the **[node]** stanza.  Here is an example configuration:
```
Comment (Allan-N) [inline docs/adv-topics/noderesolution.md:39]: for te node --> for the node

**Review by Allan-N — CHANGES_REQUESTED** [review]

**Review by Allan-N — CHANGES_REQUESTED** [review]
You also want to update .../ASL3-Manual/mkdocs.yml

File: docs/adv-topics/noderesolution.md (hunk 1 above)
Comment (Allan-N) [reply docs/adv-topics/noderesolution.md:3]: I apologize for not making my comment clear.  I was asking for a blank line between the "app_rpt has three ways ..." and the first "* An external node ...".  Without the break the list does not show up as a list.

**Review by Allan-N — CHANGES_REQUESTED** [review]

File: docs/adv-topics/noderesolution.md [hunk 6]
```diff
@@ -8,10 +9,11 @@ The above mechanisms are used to translate a node number into its respective pub
 
 app_rpt first looks to see if the node is statically configured.  If it is not found, the setting for **node_lookup_method** in the **[general]** stanza of rpt.conf determines the next step. The default setting is to perform a DNS lookup and if not found, attempt a lookup using the external file.  For more information see [HTTP Registration](https://allstarlink.github.io/adv-topics/httpreg/)
 
-*Note:  The AllStarLink network maintains security by requiring node owners to register with AllStarLink. Before a node is published to our directory, it must successfully authenticate and register with our server.* 
+*Note:  The AllStarLink network maintains security by requiring node owners to register with AllStarLink. Beforehttps://github.com/AllStarLink/asl3-update-nodelist) a node is published to our directory it must successfully authenticate and register with our server.*
```
Comment (Allan-N) [inline docs/adv-topics/noderesolution.md:12]: It looks like a URL found it's way into the text :-(

**Review by Allan-N — CHANGES_REQUESTED** [review]

**Review by tsawyer — APPROVED** [review]
Generally looks good. But did not proof read.

**Review by Allan-N — APPROVED** [review]
Looks great!

**Pull request merged**